snorkel/labeling/analysis.py

```diff
@@ -24,9 +26,12 @@ class LFAnalysis:
         See above.
     """
 
-    def __init__(self, L: np.ndarray) -> None:
+    def __init__(
+        self, L: np.ndarray, lfs: Optional[List[LabelingFunction]] = None
+    ) -> None:
         self.L = L
         self._L_sparse = sparse.csr_matrix(L + 1)
+        self._lf_names = None if lfs is None else [lf.name for lf in lfs]
```

Comment:
Check for L.shape[1] == len(lfs)?

Reply:
+1

Reply:
add check for matching LF names? in case people copy pasted the same LF twice/named two LFs the same thing

Reply:
would that be invalid?